### PR TITLE
#378 add the limitation of login

### PR DESCRIPTION
**Pull request by BelousSofiya**
I used library: https://github.com/tomasbasham/ratelimit
For now if user login 10 times in 10 minuts (no matter with wrong or right credentials) this url will be blocked for this user untill 10 minuts will finish.

### CODE REVIEW

File: forum/settings.py [hunk 1]
```diff
@@ -228,3 +229,8 @@ def running_tests():
     # For tests execution, the fastest password hasher is used
     # (may increase execution speed ×10 and more)
     PASSWORD_HASHERS = ("django.contrib.auth.hashers.MD5PasswordHasher",)
+    DELAY_FOR_LOGIN = 5
+    ATTEMPTS_FOR_LOGIN = 2
+else:
+    DELAY_FOR_LOGIN = 600  # delay time for login in seconds
+    ATTEMPTS_FOR_LOGIN = 10  # attempts for login during delay for login
```
Comment (GeorgiyII) [inline forum/settings.py:236]: Its OK for now but as for me this is bad practice to have such conditions in settings. If you want to use settings for unittests you can use @override_settings decorator

File: authentication/serializers.py [hunk 2]
```diff
@@ -2,13 +2,18 @@
 
 from django.contrib.auth import authenticate, get_user_model
 from django.core.exceptions import ValidationError
+from django.conf import settings as custom_settings
 from djoser.conf import settings
```
Comment (GeorgiyII) [inline authentication/serializers.py:6]: why not `from django.conf import settings as django_settings` and `from djoser.conf import settings as auth_settings` or `from djoser.conf import settings as djoser_settings`

This is much clearer in code when the imports says what about they are.

File: authentication/serializers.py [hunk 3]
```diff
@@ -93,18 +98,18 @@ class Meta(UserSerializer.Meta):
...
-        self.user = authenticate(
-            request=self.context.get("request"), **params, password=password
+    @RateLimitDecorator(
+        calls=custom_settings.ATTEMPTS_FOR_LOGIN,
+        period=custom_settings.DELAY_FOR_LOGIN,
+    )
+    def validate_for_rate(self, attrs):
+        email = attrs.get(settings.LOGIN_FIELD).lower()
+        new_attr = OrderedDict(
+            [("password", attrs.get("password")), ("email", email)]
```
Comment (GeorgiyII) [inline authentication/serializers.py:113]: Is there any reasons to use OrderedDict here with two keys?

File: authentication/serializers.py (hunk 2 above)
Comment (BelousSofiya) [reply authentication/serializers.py]: Thank you for feedback. I've changed names for settings to django_settings and djoser_settings

File: authentication/serializers.py (hunk 3 above)
Comment (BelousSofiya) [reply authentication/serializers.py]: Thank you for feedback. I've deleted OrderedDict

File: forum/settings.py (hunk 1 above)
Comment (BelousSofiya) [reply forum/settings.py:236]: This decorator works only if we run single testcase. Smth like this: "python manage.py test authentication.tests.test_user_login". If we run all tests at once (python manage.py test) this decorator would't work. I found this in docs: 
"Warning

The settings file contains some settings that are only consulted during initialization of Django internals. If you change them with override_settings, the setting is changed if you access it via the django.conf.settings module, however, Django’s internals access it differently. Effectively, using override_settings() or modify_settings() with these settings is probably not going to do what you expect it to do."

So, when we are running a specific test case  the override decorator takes effect. When we are running a whole suite specific testcase with override decorator probably isn't the first one that's being run and decorator wouldn't work.
The same for modify_settings() decorator. So I think it's not our varient.

**Review by GeorgiyII — APPROVED** [review]

File: forum/settings.py (hunk 1 above)
Comment (BelousSofiya) [reply forum/settings.py:236]: I'll create a task.

**Review by Lvyshnevska — APPROVED** [review]

**Pull request merged**